c++/src/ColumnWriter.cc

```diff
@@ -1194,12 +1194,11 @@ namespace orc {
     bool hasNull = false;
     for (uint64_t i = 0; i < numValues; ++i) {
       if (notNull == nullptr || notNull[i]) {
-        // TimestampVectorBatch already stores data in UTC
-        int64_t millsUTC = secs[i] * 1000 + nanos[i] / 1000000;
+        int64_t millsUTC =
+          (secs[i] + timezone.getVariant(secs[i]).gmtOffset) * 1000 + nanos[i] / 1000000;
```

Comment:
I think the original data in TimestampVectorBatch should still be UTC timestamps. So we don't need the gmtOffset adjustment here.

Reply:
You can check TimestampTreeWriter.java:113. Without this line, we will see inconsistency with java code.